### PR TITLE
WIP docs: Explain the architecture a little more

### PR DESCRIPTION
**Pull request by fabiand**
Especially highlight how TPRs, controllers, and daemons cooperate.

---
This change is [Reviewable](https://reviewable.io/reviews/kubevirt/kubevirt/46)

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -27,14 +27,87 @@ Users requiring virtualization services are speaking to the Virtualization API
...
+                                                                  |
+                                 +---------------------------+----+
+     (Cluster level)             |                           |
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+     (Node level)                |                           |
+                                 | (watch for VM changes,    |
+                                 |  on Node)                 |
+                                 |                           |
+                          +--------------+            +--------------+
+                          | virt-handler | (2)        | virt-handler | (2)
```
Comment (stu-gott) [inline docs/architecture.md]: This confuses me. It makes it feel like there's more than one virt-handler per node.

Other than that, looks great!

Comment (admiyo) [inline docs/architecture.md]: Yeah, it should have one virt hander and 2 virt-launchers.  Maybe we should rename that to virt-stub before documenting?

Comment (admiyo) [inline docs/architecture.md]: Also, show libvirt in here?

Comment (fabiand) [inline docs/architecture.md]: My main intention was to illustrate controllers (cluster wide) and daemons (host wide) and how they work with TPRs.

But it seems this needs an update to explain this better.

**Pull request merged**